### ISSUE TITLE
Delete a schedule isn't possible

### ISSUE THREAD

**Issue by BRE-MD** [ansible/awx#5137]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
If I want delete a schedule I get an error, no matter if I use the GUI or the API.
Other schedules can be deleted, only this one can't.
It's possible to change the schedule with PATCH and PUT, only DELETE isn't possible.

##### ENVIRONMENT
* AWX version: 7.0.0
* AWX install method: docker on linux
* Ansible version:  2.8.4
* Operating System: Windows
* Web Browser: Chrome

##### STEPS TO REPRODUCE
Delete the schedule. (GUI or API)
`DELETE /api/v2/schedules/22/
`

##### EXPECTED RESULTS
Deleting the schedule.

##### ACTUAL RESULTS

ERROR!
Call to /api/v2/schedules/22/ failed. DELETE returned: 500 A server error has occurred.

##### ADDITIONAL INFORMATION

194.97.51.12 - - [29/Oct/2019:13:18:51 +0000] "OPTIONS /api/v2/workflow_job_templates/ HTTP/1.1" 200 11822 "https://awx.freenet-group.de/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36" "-"
[pid: 390|app: 0|req: 24/8580] 194.97.51.12 () {50 vars in 2498 bytes} [Tue Oct 29 13:18:51 2019] OPTIONS /api/v2/workflow_job_templates/ => generated 11822 bytes in 210 msecs (HTTP/1.1 200) 10 headers in 404 bytes (1 switches on core 0)
194.97.51.12 - - [29/Oct/2019:13:18:52 +0000] "OPTIONS /api/v2/job_templates/ HTTP/1.1" 200 21596 "https://awx.freenet-group.de/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36" "-"
[pid: 341|app: 0|req: 298/8581] 194.97.51.12 () {50 vars in 2480 bytes} [Tue Oct 29 13:18:51 2019] OPTIONS /api/v2/job_templates/ => generated 21596 bytes in 540 msecs (HTTP/1.1 200) 10 headers in 404 bytes (1 switches on core 0)
[pid: 334|app: 0|req: 455/8582] 194.97.51.12 () {50 vars in 2613 bytes} [Tue Oct 29 13:18:51 2019] GET /api/v2/unified_job_templates/?page_size=20&order_by=name&type=workflow_job_template,job_template => generated 73122 bytes in 661 msecs (HTTP/1.1 200) 8 headers in 233 bytes (1 switches on core 0)
194.97.51.12 - - [29/Oct/2019:13:18:52 +0000] "GET /api/v2/unified_job_templates/?page_size=20&order_by=name&type=workflow_job_template,job_template HTTP/1.1" 200 73122 "https://awx.freenet-group.de/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36" "-"
194.97.51.12 - - [29/Oct/2019:13:18:52 +0000] "OPTIONS /api/v2/unified_job_templates/ HTTP/1.1" 200 9178 "https://awx.freenet-group.de/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36" "-"
[pid: 373|app: 0|req: 153/8583] 194.97.51.12 () {56 vars in 2621 bytes} [Tue Oct 29 13:18:52 2019] OPTIONS /api/v2/unified_job_templates/ => generated 9178 bytes in 67 msecs (HTTP/1.1 200) 8 headers in 232 bytes (1 switches on core 0)
194.97.51.12 - - [29/Oct/2019:13:18:55 +0000] "OPTIONS /api/v2/unified_jobs/ HTTP/1.1" 200 10288 "https://awx.freenet-group.de/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36" "-"
[pid: 390|app: 0|req: 25/8584] 194.97.51.12 () {56 vars in 2602 bytes} [Tue Oct 29 13:18:55 2019] OPTIONS /api/v2/unified_jobs/ => generated 10288 bytes in 75 msecs (HTTP/1.1 200) 8 headers in 233 bytes (1 switches on core 0)
[pid: 312|app: 0|req: 654/8585] 194.97.51.12 () {50 vars in 2592 bytes} [Tue Oct 29 13:18:55 2019] GET /api/v2/schedules/?page_size=20&order_by=unified_job_template__polymorphic_ctype__model => generated 20134 bytes in 164 msecs (HTTP/1.1 200) 8 headers in 239 bytes (1 switches on core 0)
194.97.51.12 - - [29/Oct/2019:13:18:55 +0000] "GET /api/v2/schedules/?page_size=20&order_by=unified_job_template__polymorphic_ctype__model HTTP/1.1" 200 20134 "https://awx.freenet-group.de/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36" "-"
194.97.51.12 - - [29/Oct/2019:13:18:55 +0000] "OPTIONS /api/v2/schedules/ HTTP/1.1" 200 13197 "https://awx.freenet-group.de/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36" "-"
[pid: 373|app: 0|req: 154/8586] 194.97.51.12 () {56 vars in 2596 bytes} [Tue Oct 29 13:18:55 2019] OPTIONS /api/v2/schedules/ => generated 13197 bytes in 94 msecs (HTTP/1.1 200) 8 headers in 239 bytes (1 switches on core 0)
194.97.51.12 - - [29/Oct/2019:13:18:55 +0000] "OPTIONS /api/v2/schedules/ HTTP/1.1" 200 13197 "https://awx.freenet-group.de/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36" "-"
[pid: 334|app: 0|req: 456/8587] 194.97.51.12 () {56 vars in 2596 bytes} [Tue Oct 29 13:18:55 2019] OPTIONS /api/v2/schedules/ => generated 13197 bytes in 97 msecs (HTTP/1.1 200) 8 headers in 239 bytes (1 switches on core 0)
RESULT 2
OKREADY
[pid: 390|app: 0|req: 26/8588] 194.97.51.12 () {34 vars in 439 bytes} [Tue Oct 29 13:19:02 2019] GET /api/v2/ping/ => generated 330 bytes in 47 msecs (HTTP/1.1 200) 8 headers in 231 bytes (1 switches on core 0)
194.97.51.12 - - [29/Oct/2019:13:19:02 +0000] "GET /api/v2/ping/ HTTP/1.1" 200 330 "-" "check_http/v2.2 (monitoring-plugins 2.2)" "-"
2019-10-29 13:19:05,855 ERROR    django.request Internal Server Error: /api/v2/schedules/22/
Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/db/backends/base/base.py", line 240, in _commit
    return self.connection.commit()
psycopg2.IntegrityError: update or delete on table "main_schedule" violates foreign key constraint "main_unifiedjob_schedule_id_766ca767_fk_main_schedule_id" on table "main_unifiedjob"
DETAIL:  Key (id)=(22) is still referenced from table "main_unifiedjob".

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib64/python3.6/contextlib.py", line 52, in inner
    return func(*args, **kwds)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/db/transaction.py", line 240, in __exit__
    connection.commit()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/db/backends/base/base.py", line 262, in commit
    self._commit()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/db/backends/base/base.py", line 240, in _commit
    return self.connection.commit()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/db/backends/base/base.py", line 240, in _commit
    return self.connection.commit()
django.db.utils.IntegrityError: update or delete on table "main_schedule" violates foreign key constraint "main_unifiedjob_schedule_id_766ca767_fk_main_schedule_id" on table "main_unifiedjob"
DETAIL:  Key (id)=(22) is still referenced from table "main_unifiedjob".

2019-10-29 13:19:05,855 ERROR    django.request Internal Server Error: /api/v2/schedules/22/
Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/db/backends/base/base.py", line 240, in _commit
    return self.connection.commit()
psycopg2.IntegrityError: update or delete on table "main_schedule" violates foreign key constraint "main_unifiedjob_schedule_id_766ca767_fk_main_schedule_id" on table "main_unifiedjob"
DETAIL:  Key (id)=(22) is still referenced from table "main_unifiedjob".

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib64/python3.6/contextlib.py", line 52, in inner
    return func(*args, **kwds)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/db/transaction.py", line 240, in __exit__
    connection.commit()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/db/backends/base/base.py", line 262, in commit
    self._commit()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/db/backends/base/base.py", line 240, in _commit
    return self.connection.commit()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/django/db/backends/base/base.py", line 240, in _commit
    return self.connection.commit()
django.db.utils.IntegrityError: update or delete on table "main_schedule" violates foreign key constraint "main_unifiedjob_schedule_id_766ca767_fk_main_schedule_id" on table "main_unifiedjob"
DETAIL:  Key (id)=(22) is still referenced from table "main_unifiedjob".

[pid: 373|app: 0|req: 155/8589] 194.97.51.12 () {54 vars in 2591 bytes} [Tue Oct 29 13:19:05 2019] DELETE /api/v2/schedules/22/ => generated 41 bytes in 100 msecs (HTTP/1.1 500) 5 headers in 185 bytes (1 switches on core 0)
194.97.51.12 - - [29/Oct/2019:13:19:05 +0000] "DELETE /api/v2/schedules/22/ HTTP/1.1" 500 41 "https://awx.freenet-group.de/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36" "-"

**Comment by rooftopcellist** [comment]
@BRE-MD I just tried to reproduce this with the latest code on a fresh install and am able to successfully delete a schedule in the UI and API.  I also checked tried making a schedule for a job template, then deleting that job template to ensure that the schedule cascade deletes with it, which it does.  

It looks like you may have a migration issue or something else causing this misleading traceback.  You might want to try running `awx-manage showmigrations` to see if all migrations have been applied.

**Comment by BRE-MD** [comment]
Yes, only one schedule is affected, other schedules can be deleted.

awx-manage showmigrations seems to be okay, here is the output:

auth
 [X] 0001_initial
 [X] 0002_alter_permission_name_max_length
 [X] 0003_alter_user_email_max_length
 [X] 0004_alter_user_username_opts
 [X] 0005_alter_user_last_login_null
 [X] 0006_require_contenttypes_0002
 [X] 0007_alter_validators_add_error_messages
 [X] 0008_alter_user_username_max_length
 [X] 0009_alter_user_last_name_max_length
 [X] 0010_alter_group_name_max_length
 [X] 0011_update_proxy_permissions
conf
 [X] 0001_initial
 [X] 0002_v310_copy_tower_settings
 [X] 0003_v310_JSONField_changes
 [X] 0004_v320_reencrypt
 [X] 0005_v330_rename_two_session_settings
 [X] 0006_v331_ldap_group_type
contenttypes
 [X] 0001_initial
 [X] 0002_remove_content_type_name
main
 [X] 0001_initial
 [X] 0002_squashed_v300_release (18 squashed migrations)
 [X] 0003_squashed_v300_v303_updates (9 squashed migrations)
 [X] 0004_squashed_v310_release (6 squashed migrations)
 [X] 0005_squashed_v310_v313_updates (3 squashed migrations)
 [X] 0006_v320_release
 [X] 0007_v320_data_migrations
 [X] 0008_v320_drop_v1_credential_fields
 [X] 0009_v322_add_setting_field_for_activity_stream
 [X] 0010_v322_add_ovirt4_tower_inventory
 [X] 0011_v322_encrypt_survey_passwords
 [X] 0012_v322_update_cred_types
 [X] 0013_v330_multi_credential
 [X] 0014_v330_saved_launchtime_configs
 [X] 0015_v330_blank_start_args
 [X] 0016_v330_non_blank_workflow
 [X] 0017_v330_move_deprecated_stdout
 [X] 0018_v330_add_additional_stdout_events
 [X] 0019_v330_custom_virtualenv
 [X] 0020_v330_instancegroup_policies
 [X] 0021_v330_declare_new_rbac_roles
 [X] 0022_v330_create_new_rbac_roles
 [X] 0023_v330_inventory_multicred
 [X] 0024_v330_create_user_session_membership
 [X] 0025_v330_add_oauth_activity_stream_registrar
 [X] 0026_v330_delete_authtoken
 [X] 0027_v330_emitted_events
 [X] 0028_v330_add_tower_verify
 [X] 0030_v330_modify_application
 [X] 0031_v330_encrypt_oauth2_secret
 [X] 0032_v330_polymorphic_delete
 [X] 0033_v330_oauth_help_text
 [X] 0034_v330_delete_user_role
 [X] 0035_v330_more_oauth2_help_text
 [X] 0036_v330_credtype_remove_become_methods
 [X] 0037_v330_remove_legacy_fact_cleanup
 [X] 0038_v330_add_deleted_activitystream_actor
 [X] 0039_v330_custom_venv_help_text
 [X] 0040_v330_unifiedjob_controller_node
 [X] 0041_v330_update_oauth_refreshtoken
 [X] 0042_v330_org_member_role_deparent
 [X] 0043_v330_oauth2accesstoken_modified
 [X] 0044_v330_add_inventory_update_inventory
 [X] 0045_v330_instance_managed_by_policy
 [X] 0046_v330_remove_client_credentials_grant
 [X] 0047_v330_activitystream_instance
 [X] 0048_v330_django_created_modified_by_model_name
 [X] 0049_v330_validate_instance_capacity_adjustment
 [X] 0050_v340_drop_celery_tables
 [X] 0051_v340_job_slicing
 [X] 0052_v340_remove_project_scm_delete_on_next_update
 [X] 0053_v340_workflow_inventory
 [X] 0054_v340_workflow_convergence
 [X] 0055_v340_add_grafana_notification
 [X] 0056_v350_custom_venv_history
 [X] 0057_v350_remove_become_method_type
 [X] 0058_v350_remove_limit_limit
 [X] 0059_v350_remove_adhoc_limit
 [X] 0060_v350_update_schedule_uniqueness_constraint
 [X] 0061_v350_track_native_credentialtype_source
 [X] 0062_v350_new_playbook_stats
 [X] 0063_v350_org_host_limits
 [X] 0064_v350_analytics_state
 [X] 0065_v350_index_job_status
 [X] 0066_v350_inventorysource_custom_virtualenv
 [X] 0067_v350_credential_plugins
 [X] 0068_v350_index_event_created
 [X] 0069_v350_generate_unique_install_uuid
 [X] 0070_v350_gce_instance_id
 [X] 0071_v350_remove_system_tracking
 [X] 0072_v350_deprecate_fields
 [X] 0073_v360_create_instance_group_m2m
 [X] 0074_v360_migrate_instance_group_relations
 [X] 0075_v360_remove_old_instance_group_relations
 [X] 0076_v360_add_new_instance_group_relations
 [X] 0077_v360_add_default_orderings
 [X] 0078_v360_clear_sessions_tokens_jt
 [X] 0079_v360_rm_implicit_oauth2_apps
 [X] 0080_v360_replace_job_origin
 [X] 0081_v360_notify_on_start
 [X] 0082_v360_webhook_http_method
 [X] 0083_v360_job_branch_override
 [X] 0084_v360_token_description
 [X] 0085_v360_add_notificationtemplate_messages
 [X] 0086_v360_workflow_approval
 [X] 0087_v360_update_credential_injector_help_text
oauth2_provider
 [X] 0001_initial
 [X] 0002_08_updates
 [X] 0003_auto_20160316_1503
 [X] 0004_auto_20160525_1623
 [X] 0005_auto_20170514_1141
 [X] 0006_auto_20171214_2232
sessions
 [X] 0001_initial
sites
 [X] 0001_initial
 [X] 0002_alter_domain_unique
social_django
 [X] 0001_initial (2 squashed migrations)
 [X] 0002_add_related_name (2 squashed migrations)
 [X] 0003_alter_email_max_length (2 squashed migrations)
 [X] 0004_auto_20160423_0400 (2 squashed migrations)
 [X] 0005_auto_20160727_2333 (1 squashed migrations)
 [X] 0006_partial
 [X] 0007_code_timestamp
 [X] 0008_partial_timestamp
sso
 [X] 0001_initial
 [X] 0002_expand_provider_options
taggit
 [X] 0001_initial
 [X] 0002_auto_20150616_2121

**Comment by AlanCoding** [comment]
Could we get a count of how many jobs reference this schedule? This should be available at `/api/v2/schedules/20/jobs/`, or via shell_plus by `UnifiedJob.objects.filter(schedule=20).count()`?

What I'm trying to diagnose is if this is a problem that arises with the number of jobs involved.

**Comment by BRE-MD** [comment]
Hi,
`/api/v2/schedules/22/jobs/` says:  `"count": 24`

**Comment by chrismeyersfsu** [comment]
@BRE-MD what kind of job is the schedule for? (I.e. Job, workflow, inventory, or project)

**Comment by elyezer** [comment]
I've tried to reproduce this with changes added by https://github.com/ansible/awx/pull/6943 and I was not able to reproduce, here is what I've done:

Created two schedules one for a JT and another for a WFJT. Both schedules were configured to run every minute and I waited at least 5 minutes to get them to run at least 5 times. Then I tried to delete the schedules, one via the schedules page and another via the schedules list on the JT's schedules tab. I was able to delete both without any issue.

With that said we can consider this as being verified.
